node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -612,13 +620,13 @@ where
 		);
 		return;
 	}
-
-	if state.requested_collations.contains_key(&(relay_parent, para_id.clone(), peer_id.clone())) {
+	let pending_collation = PendingCollation { relay_parent, para_id, peer_id };
```

Comment:
This is about as terse as the `new` constructor which exists, so I suggest we remove the `new` entirely